.github/workflows/ds_pipeline.yml

```diff
@@ -0,0 +1,43 @@
+name: Continuous Integration 
+on: 
+    push:
+        paths-ignore:
+            - 'README.md'
+
+    pull_request:
+        branches:
+            - master
+        paths-ignore:
+            - 'README.md'
+jobs:
+    job-one:
+            name: Build and test 
+            runs-on: ubuntu-latest 
+            steps: 
+            - uses: actions/checkout@v2 
+            - run: docker build --target test --tag todo-app:test .
+            - run: docker run todo-app:test
+    job-two: 
+            name: Deploy Prod
+            runs-on: ubuntu-latest
+            needs: job-one
+           # if: ${{ github.event_name == 'pull_request' && github.ref == 'refs/heads/main' }}
```

Comment:
I think you may have left it commented out by accident.

Also, `pull_request` event happens on PR events. `push` will happen whenever `main` branch is updated. It is probably the one you want